lib/protobuf/extype/timestamp.ex

```diff
@@ -1,50 +1,23 @@
-defmodule Protobuf.Extype.Timestamp do
+defimpl Extype.Protocol, for: [Google.Protobuf.Timestamp] do
   @moduledoc """
   Implement DateTime and NaiveDateTime casting for Google Timestamp.
   """
 
-  def validate_extype!(Google.Protobuf.Timestamp, "NaiveDateTime.t"), do: :naivedatetime
-  def validate_extype!(Google.Protobuf.Timestamp, "NaiveDateTime.t()"), do: :naivedatetime
-  def validate_extype!(Google.Protobuf.Timestamp, "DateTime.t"), do: :datetime
-  def validate_extype!(Google.Protobuf.Timestamp, "DateTime.t()"), do: :datetime
-  def validate_extype!(type, extype) do
+  def validate_and_to_atom_extype!(Google.Protobuf.Timestamp, "NaiveDateTime.t"), do: :naivedatetime
+  def validate_and_to_atom_extype!(Google.Protobuf.Timestamp, "NaiveDateTime.t()"), do: :naivedatetime
+  def validate_and_to_atom_extype!(Google.Protobuf.Timestamp, "DateTime.t"), do: :datetime
+  def validate_and_to_atom_extype!(Google.Protobuf.Timestamp, "DateTime.t()"), do: :datetime
+  def validate_and_to_atom_extype!(type, extype) do
     raise "Invalid extype pairing, #{extype} not compatible with #{type}. " <>
       "Supported types are DateTime.t() or NaiveDateTime.t()"
```

Comment:
can be factored out so you don't have to hard code this

Reply:
done